src/components/Resource/ResourcePage.tsx

```diff
@@ -0,0 +1,591 @@
+import {
+  DragDropContext,
+  Draggable,
+  Droppable,
+  OnDragEndResponder,
+} from "@hello-pangea/dnd";
+import { useInfiniteQuery } from "@tanstack/react-query";
+import { Link, navigate } from "raviger";
+import {
+  ReactNode,
+  RefObject,
+  Suspense,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { useTranslation } from "react-i18next";
+
+import CareIcon from "@/CAREUI/icons/CareIcon";
+
+import { Button } from "@/components/ui/button";
+
+import { ExportButton } from "@/components/Common/Export";
+import Loading from "@/components/Common/Loading";
+import BadgesList from "@/components/Resource/ResourceBadges";
+import ResourceBlock from "@/components/Resource/ResourceBlock";
+import { formatFilter } from "@/components/Resource/ResourceCommons";
+import ListFilter from "@/components/Resource/ResourceFilter";
+
+import useExport from "@/hooks/useExport";
+import useFilters from "@/hooks/useFilters";
+
+import { RESOURCE_CHOICES } from "@/common/constants";
+
+import routes from "@/Utils/request/api";
+import { callApi } from "@/Utils/request/query";
+import request from "@/Utils/request/request";
+import { QueryRoute } from "@/Utils/request/types";
+import useTanStackQueryInstead, {
+  QueryOptions,
+} from "@/Utils/request/useQuery";
+import { formatDateTime } from "@/Utils/utils";
+import { ResourceRequest } from "@/types/resourceRequest/resourceRequest";
+
+import Page from "../Common/Page";
+import { Badge } from "../ui/badge";
+import { Input } from "../ui/input";
+import { Label } from "../ui/label";
+import { ScrollArea, ScrollBar } from "../ui/scroll-area";
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../ui/select";
+import { Tabs, TabsList, TabsTrigger } from "../ui/tabs";
+
+const resourceStatusOptions = RESOURCE_CHOICES.map((obj) => obj.text);
+
+const COMPLETED = ["COMPLETED", "REJECTED"];
+const ACTIVE = resourceStatusOptions.filter((o) => !COMPLETED.includes(o));
+
+const ResourcePage = () => {
+  const {
+    qParams,
+    Pagination,
+    FilterBadges,
+    advancedFilter,
+    resultsPerPage,
+    updateQuery,
+  } = useFilters({
+    limit: 12,
+    cacheBlacklist: ["title"],
+  });
+  const [boardFilter, setBoardFilter] = useState(ACTIVE);
+  // eslint-disable-next-line
+  const appliedFilters = formatFilter(qParams);
+  const [viewMode, setViewMode] = useState<"board" | "list">("board");
```

Comment:
You should add the logic for storing in localStorage for viewMode, right now it resets on every page load.

We can create a hook to handle that in a different issue, since we will have multiple components having a similar view.

Reply:
Nah, @Mahendar0701  is taking care of that in another issue.

Reply:
Keep as is.